security/access_token.rst

```diff
@@ -0,0 +1,185 @@
+
+Using the Access Token Authenticator
+----------------------------------
+
+This guide assumes you have setup security and have created a user object
+in your application. Follow :doc:`the main security guide </security>` if
+this is not yet the case.
+
+1) Configure the Access Token Authenticator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To use the access token authenticator, you must configure a ``token_handler``.
+The token handler is a service that is able to load and verify the access token (e.g. expiration, digital signature...)
+and return the associated user identifier.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            firewalls:
+                main:
+                    header_token:
+                        token_handler: App\Security\AccessTokenHandler
+
+This handler shall implement the interface
+:class:`Symfony\\Component\\Security\\Http\\Authenticator\\AccessTokenHandlerInterface`.
+In the following example, the handler will retrieve the token from a database
+using a fictive repository.
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // src/Security/AccessTokenHandler.php
+        namespace App\Controller;
+
+        use App\Repository\AccessTokenRepository;
+        use Symfony\Component\Security\Http\Authenticator\AccessTokenHandlerInterface;
+
+        class AccessTokenHandler implements AccessTokenHandlerInterface
+        {
+            public function __construct(private readonly AccessTokenRepository $repository)
+            {
+            }
+
+            public function getUserIdentifierFrom(string $token): string
+            {
+                $accessToken = $this->repository->findOneByValue($token);
+                if ($accessToken === null || !$accessToken->isValid()) {
+                    throw new BadCredentialsException('Invalid credentials.');
+                }
+
+                return $accessToken->getUserId();
+            }
+        }
+
+.. caution::
+
+    It is important to check the token is valid.
+    For instance, in the example we verify the token has not expired.
+    With self-contained access tokens such as JWT, the handler is required to
+    verify the digital signature and understand all claims,
+    especially `iat`, `nbf` and `exp`.
+
+Important Security Considerations
+---------------------------------
+
+Because of the security weaknesses associated with the URI method,
+including the high likelihood that the URL containing the access token will be logged,
+it **SHOULD NOT** be used unless it is impossible to transport the access token in the
+request header field or the HTTP request entity-body.
+
+The request body method **SHOULD NOT** be used except in application contexts
+where participating browsers do not have access to the "Authorization" request header field.
+
+In other words: ``query_token`` and ``body_token` authenticators are not recommended.
```

Comment:
I would make it bigger with `.. caution::` bloc, wdyt?

Reply:
Correct. I will modify it.